### PR TITLE
feat(kuma-cp) add a selectors field to TrafficRoute

### PR DESCRIPTION
**Pull request by jpeach**
### Summary

To allow a TrafficRoute policy to unambiguously target a builtin gateway
Dataplane, add a `selectors` field to the resource. `selectors` is
mutually exclusive with `sources` and `destinations`, which means that
any TrafficRoute resource targets either mesh or gateway, but never both.

### Full changelog

N/A

### Issues resolved

N/A

### Documentation

N/A

### Testing

- [x] Unit tests
- [x] E2E tests
- [ ] Manual testing on Universal
- [ ] Manual testing on Kubernetes 

### Backwards compatibility

- [x] Add `backport-to-stable` label if the code is backwards compatible. Otherwise, list breaking changes.

### CODE REVIEW

File: api/mesh/v1alpha1/traffic_route.proto [hunk 1]
```diff
@@ -228,6 +215,23 @@ message TrafficRoute {
...
+      [ (validate.rules).repeated .min_items = 1, (doc.required) = true ];
+
+  // List of selectors to match services that are destinations of traffic.
+  //
+  // Notice the difference between sources and destinations.
+  // While the source of traffic is always a data plane proxy within a mesh,
+  // the destination is a service that could be either within or outside
+  // of a mesh.
+  repeated Selector destinations = 2
+      [ (validate.rules).repeated .min_items = 1, (doc.required) = true ];
```
Comment (lahabana) [inline api/mesh/v1alpha1/traffic_route.proto:229]: Seems like `doc.required` shouldn't be there anymore

Comment (lahabana) [reply api/mesh/v1alpha1/traffic_route.proto:229]: It also feels that the comment is not correct anymore. We can't really route to something outside the mesh no?

File: api/mesh/v1alpha1/traffic_route.proto [hunk 2]
```diff
@@ -228,6 +215,23 @@ message TrafficRoute {
...
+
+  // List of selectors to match builtin gateway dataplanes. If this field
+  // is given, sources and destinations must be absent (and vice versa).
+  repeated Selector selectors = 4 [ (validate.rules).repeated .min_items = 1 ];
```
Comment (lahabana) [inline api/mesh/v1alpha1/traffic_route.proto:233]: So I guess this is a "source selector" and the destination is "*" always when using a gateway?

**Comment by jpeach** [comment]
Abandoning this PR, since we agreed to use GatewayRoute for gateway and TrafficRoute for mesh.

**Pull request closed without merging**